hs/src/Reach/Eval/Core.hs

```diff
@@ -2412,11 +2412,11 @@ evalApplyVals rator randvs =
     SLV_Clo clo_at sc -> evalApplyClosureVals clo_at sc randvs
     SLV_CloTyped clo_at sc tf -> do
       at <- withAt id
-      let isDLVar = \case { SLV_DLVar _ -> True; _ -> False } . snd
-      let ct = if all isDLVar randvs then CT_Assume True else CT_Assert
+      m <- readSt st_mode
+      let ct = if m == SLM_Module then CT_Assume True else CT_Assert
```

Comment:
I feel like such a pain... but what about:

```
const f_impl = (x) => x + 1;
const f = is(f_impl, Refine(Fun([UInt], UInt), ([x] => x > 1), (_ _ => true)));
const x = f(0);
```

This should fail, because `0` is not `> 1` and this is at the module level, but I think your code will assume it rather than assert it

Reply:
oh duh, thanks for catching that. I should have checked both: `all isDLVar randvs && mode == SLM_Module`

Reply:
```
const y = forall(UInt);
const x = f(y);
````

This should also fail, because we can't prove that `y` is `> 1`

Reply:
I think there needs to be "another way" to know that you are exporting. Like `SLM_Exporting` that gets switched to `SLM_Module` on the application or something like that (i.e. that only unravels once)

Reply:
Another option would be for the exporting code to see through `evalApplyVals` and pass it an extra arg or something

Reply:
Okay everything works with using `SLM_Exporting` to define the exported functions and when applying a typed closure in `Exporting` mode, it switches to `SLM_Module` temporarily to statically check it's arguments. Pushing up..